app/frontend/src/features/communities/events/EventPage.tsx

```diff
@@ -0,0 +1,211 @@
+    display: "grid",
+    gridTemplateAreas: `
+      "backButton eventTitle eventTitle"
+      "eventTime eventTime eventTime"
+      "attendanceButton attendanceButton ."
+    `,
+    gridAutoFlow: "column",
+    gridTemplateColumns: "3.125rem 1fr auto",
+    marginBlockEnd: theme.spacing(4),
+    marginBlockStart: theme.spacing(2),
+    [theme.breakpoints.up("sm")]: {
+      gridTemplateAreas: `
+      "backButton eventTitle attendanceButton"
+      ". eventTime eventTime"
+    `,
+    },
+  },
+  backButton: {
+    gridArea: "backButton",
+    width: "3.125rem",
+    height: "3.125rem",
+  },
+  eventTitle: {
+    gridArea: "eventTitle",
+  },
+  attendanceButton: {
+    gridArea: "attendanceButton",
+    justifySelf: "start",
+  },
+  eventTypeText: {
+    color: theme.palette.grey[600],
+  },
+  eventTimeContainer: {
+    gridArea: "eventTime",
+    display: "grid",
+    columnGap: theme.spacing(1),
+    gridTemplateColumns: "3.75rem auto",
+  },
+  calendarIcon: {
+    marginInlineStart: theme.spacing(-0.5),
+    height: "3.75rem",
+    width: "3.75rem",
+  },
+  eventDetailsContainer: {
+    display: "grid",
+    rowGap: theme.spacing(3),
+    marginBlockEnd: theme.spacing(5),
+  },
+  cardSection: {
+    padding: theme.spacing(2),
+    "& + &": {
+      marginBlockStart: theme.spacing(3),
+    },
+  },
+}));
+
+export default function EventPage() {
+  const classes = useEventPageStyles();
+  const history = useHistory();
+  const { eventId: rawEventId, eventSlug } =
+    useParams<{ eventId: string; eventSlug?: string }>();
+
+  const eventId = +rawEventId;
+  const isValidEventId = !isNaN(eventId) && eventId > 0;
+  const queryClient = useQueryClient();
+  const {
+    data: event,
+    error: eventError,
+    isLoading,
+  } = useQuery<Event.AsObject, GrpcError>({
+    queryKey: eventKey(eventId),
+    queryFn: () => service.events.getEvent(eventId),
+    enabled: isValidEventId,
+  });
+
+  const {
+    isLoading: isSetEventAttendanceLoading,
+    error: setEventAttendanceError,
+    mutate: setEventAttendance,
+  } = useMutation<Event.AsObject, GrpcError, AttendanceState>(
+    (currentAttendanceState) => {
+      const attendanceStateToSet =
+        currentAttendanceState === AttendanceState.ATTENDANCE_STATE_GOING
+          ? AttendanceState.ATTENDANCE_STATE_NOT_GOING
+          : AttendanceState.ATTENDANCE_STATE_GOING;
+      return service.events.setEventAttendance({
+        attendanceState: attendanceStateToSet,
+        eventId,
+      });
+    },
+    {
+      onSuccess(updatedEvent) {
+        queryClient.setQueryData<Event.AsObject>(
+          eventKey(eventId),
+          updatedEvent
+        );
+        queryClient.invalidateQueries(eventKey(eventId));
+        queryClient.invalidateQueries({
+          predicate: ({ queryKey }) =>
+            queryKey[0] === eventAttendeesBaseKey &&
+            isEventUsersInputType(queryKey[1]) &&
+            queryKey[1].eventId === eventId,
+        });
+      },
+    }
+  );
```

Comment:
This is not working as intended at the moment... when you flick between "going" and "not going", a new user shows up in the attendees list instead of the current user.

I can't see how the frontend code has messed up here so likely a bug in the backend? 🤷

Reply:
Lol this is so much fun, keep clicking and cycle through users 😆